src/google-social-provider.js

```diff
@@ -0,0 +1,103 @@
+  console.log('GoogleSocialProvider called');
+  this.dispatchEvent_ = dispatchEvent;
+  this.networkName_ = 'google';
+  this.initState_();
+};
+GoogleSocialProvider.prototype = new FirebaseSocialProvider();
+
+GoogleSocialProvider.prototype.getOAuthToken_ = function() {
+  var OAUTH_REDIRECT_URLS = [
+    "https://fmdppkkepalnkeommjadgbhiohihdhii.chromiumapp.org/",
+    "https://www.uproxy.org/oauth-redirect-uri",
+    "http://freedomjs.org/",
+    'http://localhost:8080/'
+  ];
+  var OAUTH_CLIENT_ID = '746567772449-jkm5q5hjqtpq5m9htg9kn0os8qphra4d.apps.googleusercontent.com';
+  var OAUTH_SCOPE = 'https://www.googleapis.com/auth/plus.login%20https://www.googleapis.com/auth/plus.me%20https://www.googleapis.com/auth/userinfo.email%20https://www.googleapis.com/auth/userinfo.profile';
+
+  var oauth = freedom["core.oauth"]();
+  return oauth.initiateOAuth(OAUTH_REDIRECT_URLS).then(function(stateObj) {
+    var oauthUrl = "https://accounts.google.com/o/oauth2/auth?" +
+        "client_id=" + OAUTH_CLIENT_ID +
+        "&scope=" + OAUTH_SCOPE +
+        "&redirect_uri=" + encodeURIComponent(stateObj.redirect) +
+        "&state=" + encodeURIComponent(stateObj.state) +
+        "&response_type=token";
+    var url = 'https://accounts.google.com/accountchooser?continue=' +
+        encodeURIComponent(oauthUrl);
+    return oauth.launchAuthFlow(url, stateObj);
+  }).then(function(responseUrl) {
+    return responseUrl.match(/access_token=([^&]+)/)[1];
+  }).catch(function (err) {
+    return Promise.reject('Login error: ' + err.message);
+  });
+};
+
+GoogleSocialProvider.prototype.loadUsers_ = function() {
+  // TODO: should we periodically check for new friends?  Or just force
+  // users to logout then login again to detect new friends?
+  this.googleGet_('plus/v1/people/me/people/visible').then(function(resp) {
+    console.log('got my contacts, ' + JSON.stringify(resp));
+    for (var i = 0; i < resp.items.length; ++i) {
+      var friend = resp.items[i];
+      this.addUserProfile_({
+        userId: friend.id,
+        name: friend.displayName,
+        imageData: friend.image.url,
+        url: friend.url
+      });
+    }
+  }.bind(this)).catch(function(e) {
+    console.error('Error loading Google users', e);
+  });
+};
+
+
+GoogleSocialProvider.prototype.getMyUserProfile_ = function() {
+  if (!this.loginState_) {
+    throw 'Error in GoogleSocialProvider.getMyUserProfile_: not logged in';
+  }
+  var cachedUserProfile =
+      this.loginState_.authData[this.networkName_].cachedUserProfile;
+  return {
+    userId: this.getUserId_(),
+    name: cachedUserProfile.name,
+    lastUpdated: Date.now(),
+    url: cachedUserProfile.link,
+    imageData: cachedUserProfile.picture + '?sz=50'
+  };
+};
+
+
+GoogleSocialProvider.prototype.googleGet_ = function(endPoint) {
+  if (!this.loginState_) {
+    throw 'Not signed in';
+  }
+  var xhr = new XMLHttpRequest();
+  // TODO: is it safe to include the API KEY?
```

Comment:
I think it's worth resolving this TODO for this pull request, as well as double-checking similar things for e.g. OAUTH_CLIENT_ID and some of the other hardcoded values/urls.